src/cascadia/TerminalConnection/ConnectionInformation.cpp

```diff
@@ -0,0 +1,61 @@
+#include "pch.h"
+#include "ConnectionInformation.h"
+#include "ConnectionInformation.g.cpp"
+
+namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
+{
+    ConnectionInformation::ConnectionInformation(hstring const& className,
+                                                 const Windows::Foundation::Collections::ValueSet& settings) :
+        _ClassName{ className },
+        _Settings{ settings }
+    {
+    }
+
+    // Function Description:
+    // - Create an instance of the connection specified in the
+    //   ConnectionInformation, and Initialize it.
+    // - This static method allows the content process to create a connection
+    //   from information that lives in the window process.
+    // Arguments:
+    // - info: A ConnectionInformation object that possibly lives out-of-proc,
+    //   containing the name of the WinRT class we should activate for this
+    //   connection, and a bag of setting to use to initialize that object.
+    // Return Value:
+    // - <none>
+    TerminalConnection::ITerminalConnection ConnectionInformation::CreateConnection(TerminalConnection::ConnectionInformation info)
+    try
+    {
+        Windows::Foundation::IInspectable inspectable{};
+
+        auto name = static_cast<HSTRING>(winrt::get_abi(info.ClassName()));
+        auto pointer = winrt::put_abi(inspectable);
+        ::IInspectable** raw = reinterpret_cast<::IInspectable**>(pointer);
```

Comment:
this seems scary. is there not a way to just get an abi-compatible `IInspectable**` directly?

Reply:
> mfw you suggested that I do this in the first place

![image](https://user-images.githubusercontent.com/18356694/126213672-e5cc8dd1-b98e-48a0-b381-382fbbf27c79.png)

Reply:
Right right, but I meant something closer to, "in the intervening time, did we find something that is less ~ ~ weird ~ ~ than that?" :smile:

(reinterpret_casting a pointer is scary, and it didn't seem necessary from the original example :smile:)